index.html

```diff
@@ -693,14 +706,93 @@ <h4>Proof Verification (ecdsa-2019)</h4>
+        <h3>Choice of ECDSA and Parameters</h3>
+        <p>The ECDSA signature scheme has the <strong>EUF-CMA</strong> 
+(<em>existential unforgeability under chosen message attacks</em>) security 
+property. This property guarantees that any efficient adversary who has the 
+public key <em>pk</em> of the signer and received an arbitrary number of 
+signatures on messages of its choice (in an adaptive manner) cannot output a 
+valid signature for a new message (except with negligible probability).
+        </p>
+        <p><strong>SUF-CMA</strong> (<em>strong unforgeability under chosen 
+message attacks</em>) is a stronger notion than <em>EUF-CMA</em>. It guarantees 
+that for any efficient adversary who has the public key <em>pk</em> of the 
+signer and received an arbitrary number of signatures on messages of its 
+choice, it cannot output a new valid signature pair for either a new message 
+or a new signature for an old message (except with negligible probability). 
+ ECDSA signature scheme does <strong>not</strong> have the SUF-CMA property, 
+ while other schemes such as EdDSA [[FIPS-186-5]] do.
+        </p>
+        <p>Per [[NIST-SP-800-57-Part-1]] in the absence of large scale quantum 
+computers a <em>security strength</em> level of 128 bits requires a key size 
+of approximately 256 bits while a security strength level of 192 bits requires 
+a key size of 384 bits. [[NIST-SP-800-186]] recommendations includes curves 
+P-256 and P-384 at these security strength levels respectively.
+        </p>
+      </section>
+      <section class="informative">
+        <h3>Implementation Considerations for ECDSA Algorithms</h3>
+        <p>The ECDSA algorithm as detailed in [[FIPS-186-5]] states: &quot;A 
+new secret random number <em>k</em>, shall be generated prior to the generation 
+of each digital signature for use during the signature generation process.&quot; 
+The failure to properly generate this <em>k</em> value has lead to some highly 
+public, integrity breaches in widely deployed systems. To counter this problem 
+a hash based method of determining the secret number <em>k</em>, called 
+<em>Deterministic ECDSA</em>, is given in [[FIPS-186-5]] and [[RFC6979]]. 
+Verification of a ECDSA signature is independent of the method of generating 
+<em>k</em>. Hence it is generally recommended to use <em>Deterministic 
+ECDSA</em> unless other requirements dictate otherwise.
+        </p>
+      </section>
+      <section class="informative">
+        <h3>Key Management</h3>
+        <p>The security of the ECDSA algorithm is dependent on the quality and 
+protection of its <em>private signing key</em>. Guidance in the management of 
+cryptographic keys is a large subject and the reader is referred to 
+[[NIST-SP-800-57-Part-1]] for more extensive recommendations and discussion. 
+As mentioned in both [[FIPS-186-5]] and [[NIST-SP-800-57-Part-1]] an ECDSA 
+private signing key should <strong>not</strong> be used for any other purpose 
```

Comment:
Avoid RFC "should", "must", and "may" language, even if lower cased, as reviewers get confused and tend to ask if we really meant SHOULD. Try alternative english language like "is expected to"... "it is strongly advised that"... "might" and so on.

Reply:
Fixed. Thanks for wording recommendations.